server/services/chat/olly_chat_service.ts

```diff
@@ -21,22 +25,89 @@ const INTERACTION_ID_FIELD = 'parent_interaction_id';
 
 export class OllyChatService implements ChatService {
   static abortControllers: Map<string, AbortController> = new Map();
+  private static rootAgentId: string | undefined;
 
-  private async requestAgentRun(
-    rootAgentId: string,
-    payload: AgentRunPayload,
-    context: RequestHandlerContext
-  ) {
+  constructor(
+    private readonly context: RequestHandlerContext,
+    private readonly rootAgentName: string
+  ) { }
+
+  private async initRootAgent() {
+    OllyChatService.rootAgentId = await this.searchRootAgent(this.rootAgentName);
```

Comment:
The `initRootAgent` will be called during the `searchRootAgent` calling the ml-commons API. That will cause unnecessary `searchRootAgent` called. Shall we store the first `searchRootAgent` promise and reuse it?

Reply:
`rootAgentId` is static, it's set when the first call triggers and will not be reset unless encounters a 404 error, we have this check below:
```
if (!OllyChatService.rootAgentId) {
      await this.initRootAgent();
      this.context.assistant_plugin.logger.info(
        `root agent id has not been initialized yet, init it at the first time, current root agent id is:${OllyChatService.rootAgentId}`
      );
    }
```

Reply:
The `OllyChatService.rootAgentId` will still be empty If the first request not response , right? If we call the `requestLLM` or `regenerate` at that time, the  `OllyChatService.rootAgentId` is still empty and will call `initRootAgent` again. This's not a big issue. I'm open to add todo here and fixed it in the future version.

Reply:
Gotcha, thanks, I'll think about how to implement that.

Reply:
Will do this in next PR, thanks!